src/MainWindow.vala

```diff
@@ -154,11 +154,15 @@ public class AppCenter.MainWindow : Gtk.ApplicationWindow {
         updates_badge = new Gtk.Label ("!");
         updates_badge.get_style_context ().add_class ("badge");
 
+        var eventbox_badge = new Gtk.EventBox ();
+        eventbox_badge.add (updates_badge);
+        eventbox_badge.button_press_event.connect (badge_event);
```

Comment:
It's better to use the release event here so that we don't respond to things like drag events if the user wants to move the window for example

```suggestion
        eventbox_badge.button_release_event.connect (badge_event);
```